src/main/java/training/bank/Account.java

```diff
@@ -1,30 +1,34 @@
-package training.supportbank;
+package training.bank;
 
+import java.lang.reflect.Array;
+import java.math.BigDecimal;
+import java.util.ArrayList;
 import java.util.HashSet;
+import java.util.List;
 
 public class Account {
     private String owner;
-    private double balance;
+    private BigDecimal balance = BigDecimal.ZERO;
     private HashSet<Transaction> debts = new HashSet<>();
```

Comment:
Is there a reason you've chosen a `HashSet` here? It adds some complexity over using a `List`. Normally the only advantage a `HashSet` has over a `List` is that you can answer the question "is this particular element a member of the set" very quickly - but I don't think you do that here?